File: B_Move_To_The_End.py
# idea is take the last k-1 sum,
# with the remaining k, find the max number you can get,
# go back now add it to your num , pop that from the heap,
# repeat

import sys
from collections import Counter, defaultdict
import heapq
from bisect import bisect_left, bisect_right
from copy import deepcopy
import math

# input = input
input = sys.stdin.readline


def solve():
    n = int(input())
    nums = list(map(int, input().split()))
    _maxs = []
    curr = 0
    for num in nums:
        curr = max(curr, num)
        _maxs.append(curr)

    out = []
    _sum = 0
    for i in range(n - 1, -1, -1):
        out.append(_sum + _maxs[i])
        _sum += nums[i]
    print(*out)


def main():
    t = int(input())

    for _ in range(t):
        solve()


if __name__ == "__main__":
    main()

File: B_Move_To_The_End.cpp
#include <bits/stdc++.h>
using namespace std;
template <typename T> T getMax(const std::vector<T> &nums) {
    return *max_element(nums.begin(), nums.end());
}

void solve() {
    int n;
    cin >> n;
    vector<long long> nums(n);
    for (long long &x : nums) {
        cin >> x;
    }
    vector<long long> _maxs(n);
    long long curr = 0;
    for (int i = 0; i < n; i++) {
        curr = max(curr, nums[i]);
        _maxs[i] = curr;
    }
    long long _sum = 0;
    for (int i = n - 1; i > -1; i--) {
        cout << _maxs[i] + _sum << " ";
        _sum += nums[i];
    }
    cout << "\n";
}
int main() {

    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t = 1;
    cin >> t;
    while (t--) {
        solve();
    }
}
